R/optionalInput.R

```diff
@@ -101,14 +173,15 @@ optionalSelectInput <- function(inputId, # nolint
   )
   checkmate::assert_flag(multiple)
   checkmate::assert_string(sep, null.ok = TRUE)
-  stopifnot(is.list(options))
+  checkmate::assert_list(options)
   checkmate::assert(
     checkmate::check_string(label_help, null.ok = TRUE),
     checkmate::check_class(label_help, "shiny.tag"),
     checkmate::check_class(label_help, "shiny.tag.list"),
     checkmate::check_class(label_help, "html")
   )
```

Comment:
we could create a helper class `called check_char_html `that checks for HTML characters and use it to validate `label` and `label_help ` .
```
check_char_html <- function(val) {
  checkmate::assert(
    checkmate::check_string(val, null.ok = TRUE),
    checkmate::check_multi_class(val, c("shiny.tag", "shiny.tag.list", "html"))
}
```

and 
```
check_char_html(label)
check_char_html(label_help)
```

Reply:
I wish to do this but this function probably needs some `deparse(val)` so the `checkmate::assert` throws meaningful message (see `.var.name` in the arguments). I'd prefer to postpone such decisions for later (after release)

Reply:
Why not just run 
```
  checkmate::assert(
    checkmate::check_string(val, null.ok = TRUE),
    checkmate::check_multi_class(val, c("shiny.tag", "shiny.tag.list", "html")
  )
```